craft_parts/ctl.py

```diff
@@ -0,0 +1,75 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2017-2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Helpers to invoke step execution handlers from the command line."""
+
+import json
+import os
+import sys
+from typing import List
+
+
+def main():
+    """Run the ctl client cli."""
+    if len(sys.argv) < 2:
+        print(f"usage: {sys.argv[0]} <command>")
+        sys.exit(1)
+
+    cmd, param = sys.argv[1], sys.argv[2:]
+    try:
+        client(cmd, param)
+    except RuntimeError as err:
+        print(f"Error: {err}.", file=sys.stderr)
```

Comment:
I don't think it is desirable to pollute output with this extra text.
```suggestion
        logger.debug("ctl error: %r", err)
```

Reply:
I'll set the logger level to info so the error message is actually shown if partsctl is executed from the command line. This should be replaced by the appropriate standard messages call later.
```
$ partsctl pull
ctl error: 'PARTS_CALL_FIFO' environment variable must be defined.
Note that this utility is designed for use only in part scriptlets.
```

Reply:
Thanks!